### ISSUE TITLE
RefVariable and NDArray interaction

### ISSUE THREAD

**Issue by sfxrk** [SciSharp/TensorFlow.NET#203]
python code is direct and simple:
import tensorflow as tf
import numpy as np

x_data = np.random.rand(100)  # .astype(np.float32)
y_data = 0.1 * x_data + 0.2
k = tf.Variable(0.)  # 1.1
b = tf.Variable(0.)  # 0.5
y = k * x_data + b
print(type(y))

# 二次代价函数
loss = tf.reduce_mean(tf.square(y_data - y))
# 梯度下降进行训练的优化器
optimizer = tf.train.GradientDescentOptimizer(0.2)
# 最小化二次代价函数
train = optimizer.minimize(loss)
init = tf.global_variables_initializer()
with tf.Session() as sess:
    # 初始化变量
    sess.run(init)
    for step in range(201):
        sess.run(train)
        if step % 20 == 0:
            print(step, sess.run([k, b]))

but with tf.net 
        public static void tf2_4linear()
        {
            NDArray x_data = np.random.randn(100);
            NDArray y_data = 0.1 * x_data + 0.2;
            RefVariable k = tf.Variable(0.0f);
            RefVariable b = tf.Variable(0.0f);
            int n = x_data.size;
            // python ok, how to translate to tfnet?
            //  y_python = k * x_data + b;
            var y = x_data.ToMuliDimArray<float>();//ToMultiDimArray ?
            var y1 = k * x_data + b;// compile error:RefVariable, NDArray
            var y2 = tf.add(tf.multiply(k, x_data), b);//compile error:NDArray, Tensor
            var y4 = tf.add(k * x_data, b);// compile error:RefVariable, NDArray
            var y5 = tf.add(x_data * k, b);// compile error:NDArray, RefVariable
            //second loss function
            var loss1 = tf.reduce_mean(square(y_data - y));
            Tensor loss2 = tf.reduce_sum(tf.pow(y - y_data, 2)) / (2.0f * n);
            //optimizer
            var optimizer = tf.train.GradientDescentOptimizer(0.2f);
            //minize
            var train = optimizer.minimize(loss);
            Operation init = tf.global_variables_initializer();
            using (Session sess = tf.Session())
            {
                sess.run(init);
                for (int step = 0; step < 201; step++)
                {
                    sess.run(train);
                    if (step % 20 == 0)
                    {
                        //var fetch = { k, b };
                        Console.WriteLine(step.ToString() + " "
                            + sess.run(k).ToString() + " "
                            + sess.run(b).ToString());

                    }
                }
            }

        }
also , if make variable to tensor,also runtime error:
        public static void tf2_4linear2()
        {
            NDArray x_data = np.random.randn(100);                                //np.random.rand(100)
            NDArray y_data = 0.1 * x_data + 0.2;
            Tensor x_data2 = tf.constant(x_data);
            Tensor y_data2 = tf.constant(y_data);
            RefVariable k = tf.Variable(0.0f);
            RefVariable b = tf.Variable(0.0f);

            int n = x_data.size;
            //var y = x_data.ToMuliDimArray<float>();//ToMultiDimArray ?
            Tensor y1 = k * x_data2 + b;
            //NumSharp.Core.IncorrectShapeException: 
            //'This method does not work with this shape or was not already implemented.'
            var loss1 = tf.reduce_mean(square(y_data - y1));
            //System.NullReferenceException: 'Object reference not set to an instance of an object.'
            Tensor loss2 = tf.reduce_sum(tf.pow(y1 - y_data2, 2)) / (2.0f * n);
            //optimizer
            var optimizer = tf.train.GradientDescentOptimizer(0.2f);
            //minize
            var train = optimizer.minimize(loss2);
            Operation init = tf.global_variables_initializer();
            using (Session sess = tf.Session())
            {
                sess.run(init);
                for (int step = 0; step < 201; step++)
                {
                    sess.run(train);
                    if (step % 20 == 0)
                    {
                        //var fetch = { k, b };
                        Console.WriteLine(step.ToString() + " " + sess.run(k).ToString() + " " + sess.run(b).ToString()); 
                    }
                }
            } 
        }
